chemprop/args.py

```diff
@@ -123,6 +124,128 @@ class CommonArgs(Tap):
+    """
+    Mode_of_contrastive_learning_probabilities: List = [1/6, 1/6, 1/6, 1/6, 1/6, 1/6]
+    """
+    For self supervised pretraining the contrastive learning different modes probabilities.
+    """
+    is_triplet: bool = False
+    """
+    For self supervised pretraining whether the triplet loss that has three views as input is added
+    """
+    triplet_margin: float = 0.0
+    """
+    For self supervised pretraining, the triplet loss margin parameter
+    """
+    triplet_p: int = 2
+    """
+    For self supervised pretraining, the triplet loss p parameter
+    """
+    triplet_loss_para: float = 0.1
+    """
+    For self supervised pretraining, the trade-off hyperparameter
+    """
+    MA_first_percent: float = 0.1
+    """
+    For self supervised pretraining, the mask atom percentage for first view 
+    """
+    MA_second_percent: float = 0.3
+    """
+    For self supervised pretraining, the mask atom percentage for second view
+    """
+    BD_first_percent: float = 0.1
+    """
+    For self supervised pretraining, the mask bond percentage for first view 
+    """
+    BD_second_percent: float = 0.3
+    """
+    For self supervised pretraining, the mask bond percentage for second view
+    """
+    SG_first_percent: float = 0.1
+    """
+    For self supervised pretraining, the subgraph deletion percentage for first view 
+    """
+    SG_second_percent: float = 0.3
+    """
+    For self supervised pretraining, the subgraph deletion percentage for second view
+    """
+    pretrain_with_val: bool = False
+    """
+    For self supervised pretraining, whether the model is only saved with the validation score lower, not implemented well yet
+    """
+    pretrain_save_per_epoch: int = 2
+    """
+    For self supervised pretraining,save model for number of epoch trained. Normally, big model is only saved with this kind of criteria 
+    """
+    is_whole_finetune_pretrain_checkpoint: bool = False
+    """
+    Whether to load a SSL pretrained model and fit it to a new model structure and finetune the whole model 
+    """
+    large_mpn_encoder: bool = False
+    """
+    Whether to train a large mpn encoder
+    """
+    use_resnet: bool = False
+    """
+    Whether to use resnet structure for message passing
+    """
+    depth_resnet: bool = False
+    """
+    Whether to use resnet structure for message passing for each depth
+    """
+    per_depth_mpn_number: int = 1
+    """
+    The number of FFNs in each depth for the mpn
+    """
+    DDP_training: bool = False
+    """
+    Whether to use DDP training or not, necessary for large data and large model. Now it is only implemented for SSL_pretrain
+    """
+    DDP_rank: int = 0
+    """
+    The current rank information, it will be set dynamically for different processes
+    """
+    load_DDP: bool = False
+    """
+    Whether to load from a DDP trained model, since it will have module prefix.
```

Comment:
If DDP-trained models have a different prefix, can't we just deduce that they are DDP models by checking for said prefix (making this arg unnecessary)?